File: DotNetGotchas/CSharp/NotToFinalize/FinalizeAndManagedResources/SomeOtherClass.cs
//SomeOtherClass.cs
using System;

namespace FinalizeAndManagedResources
{
	public class SomeOtherClass
	{
		private int[] values = new int[1000000];

		public void Notify()
		{
			Console.WriteLine("Notify called on SomeOtherClass");
		}

		~SomeOtherClass()
		{
			Console.WriteLine("*** SomeOtherClass Finalized");
		}
	}
}

File: DotNetGotchas/VBDN/NotToFinalize/FinalizeAndManagedResources/SomeOtherClass.vb
'SomeOtherClass.vb
Public Class SomeOtherClass
    Private values() As Integer = New Integer(1000000 - 1) {}

    Public Sub Notify()
        Console.WriteLine("Notify called on SomeOtherClass")
    End Sub


    Protected Overrides Sub Finalize()
        MyBase.Finalize()
        Console.WriteLine("*** SomeOtherClass Finalized")
    End Sub
End Class